Devops/fairspc 55 GitHub actions

tgreenwood opened the pull request.

tgreenwood committed:
Added github actions workflow to package and push helm chart

tgreenwood committed:
Removed Travis's scripts and updated README.md

ewelinagr commented:
hmm, so there will be no check on PR? I would be good to see if the packaging is not broken or e.g. there are no lint errors (it was there on travis build, although I see the build was passing even with linter errors...)
https://api.travis-ci.com/v3/job/612025435/log.txt

tgreenwood commented:
> hmm, so there will be no check on PR? It would be good to see if the packaging is not broken or e.g. there are no lint errors (it was there on travis build, although I see the build was passing even with linter errors...) https://api.travis-ci.com/v3/job/612025435/log.txt

@ewelinagr good point, added linter and realized that it would be easier to understand what exactly fails if put all helm commands into separate steps, so you will find more steps. 

By the way, GH Actions fails even if there is a warning

frankyhollywood commented on charts/fairspace-keycloak/values.yaml:
```diff
@@ -20,6 +20,7 @@ fairspaceKeycloak:
         cpu: "500m"
         memory: "2048Mi"
   postgresql:
+    password: ""
```
Is there a reason to add empty password? Rather have a failing deployment, than default password ""

tgreenwood replied:
the only reason is to satisfy helm linter's complains

tgreenwood replied:
let me see what else we can do

tgreenwood replied:
okay, I've done it in a different way: it did not fail before my changes because of the default value in pipeline, but linter failed. I made it behaving the same adding brackets what allows to have nullable values. In this case, we will take the default value again if it is absent

tgreenwood replied:
well, it does not improve, but at least it does not make it worse. I wouldn't spend time on that

tgreenwood committed:
resolved helm linter complain

ewelinagr commented:
> > hmm, so there will be no check on PR? It would be good to see if the packaging is not broken or e.g. there are no lint errors (it was there on travis build, although I see the build was passing even with linter errors...) https://api.travis-ci.com/v3/job/612025435/log.txt
> 
> @ewelinagr good point, added linter and realized that it would be easier to understand what exactly fails if put all helm commands into separate steps, so you will find more steps.
> 
> By the way, GH Actions fails even if there is a warning

So there is no easy way to have some build/action on every PR?

tgreenwood committed:
Made the push to repository step conditional (only for dev and release branches)

tgreenwood commented:
> > > hmm, so there will be no check on PR? It would be good to see if the packaging is not broken or e.g. there are no lint errors (it was there on travis build, although I see the build was passing even with linter errors...) https://api.travis-ci.com/v3/job/612025435/log.txt
> > 
> > 
> > @ewelinagr good point, added linter and realized that it would be easier to understand what exactly fails if put all helm commands into separate steps, so you will find more steps.
> > By the way, GH Actions fails even if there is a warning
> 
> So there is no easy way to have some build/action on every PR?

@ewelinagr good point, it makes sense to know that something is incorrect before merge. Changed the event triggerning (now on any push) and the push now is conditional (for dev and releases branches only
